### DISCUSSION TITLE
crash under android platform when qt use the  native library

### DISCUSSION THREAD

**Discussion by yygbruce** [google/filament#6250] [General]
I use the filament's native library with Qt5.12 develp an app, it run well under ubuntu 18.04;but under the android platform, it crash.I found that these codes cause the crash.
if (renderer_->beginFrame(swap_chain_)) {
    renderer_->render(view_);//this line
    renderer_->endFrame();//this line
    engine_->flushAndWait();//this line
  }
the swap_chain_ is created by native window handle, has anyone encountered a similar problem?

**Comment by romainguy** [comment]
That's not enough information. How do you create the swap chain? And are you using Qt on Android?

**Reply by yygbruce** [reply]
I AdrenoGLES-0: QUALCOMM build : 9b824c0bed, I6c7bff0fbd
I AdrenoGLES-0: Build Date : 10/09/21
I AdrenoGLES-0: OpenGL ES Shader Compiler Version: EV031.32.02.15
I AdrenoGLES-0: Local Branch :
I AdrenoGLES-0: Remote Branch :
I AdrenoGLES-0: Remote Branch :
I AdrenoGLES-0: Reconstruct Branch :
I AdrenoGLES-0: Build Config : S P 10.0.7 AArch64
I AdrenoGLES-0: Driver Path : /vendor/lib64/egl/libGLESv2_adreno.so
I AdrenoGLES-0: PFP: 0x016ee190, ME: 0x00000000
D hw-ProcessState: Binder ioctl to enable oneway spam detection failed: Invalid argument
E project.exampl: open libmigui.so failed! dlopen - dlopen failed: library "libmigui.so" not found
I Filament: FEngine (64 bits) created at 0x71052ee000 (threading is enabled)
D Filament: Using ASurfaceTexture
I Filament: FEngine resolved backend: OpenGL
D DecorView[]: onWindowFocusChanged hasWindowFocus true
V Filament: [Qualcomm], [Adreno (TM) 618], [OpenGL ES 3.2 V@0502.0 (GIT@9b824c0bed, I6c7bff0fbd, 1633761551) (Date:10/09/21)], [OpenGL ES GLSL ES 3.20]
V Filament: Feature level: 2
V Filament: Active workarounds:
V Filament: dont_use_timer_query
V Filament: disable_sidecar_blit_into_texture_array
V Filament: split_easu
V Filament: invalidate_end_only_if_invalidate_start
V Filament: allow_read_only_ancillary_feedback_loop
V Filament: enable_initialize_non_used_uniform_array
I Filament: FEngine feature level: 2
F libc : Fatal signal 11 (SIGSEGV), code 1 (SEGV_MAPERR), fault addr 0x91 in tid 11168 (FEngine::loop), pid 11113 (project.example)
11:18:32:

"org.qtproject.example" died.

**Reply by yygbruce** [reply]
sample viewtest.cpp

**Comment by yygbruce** [comment]
Ok, I upgraded the qt's verson from 5.12.12 to 5.15.2, and it can work.Hope this is useful for somebody.
